### PR TITLE
Use Kotlin 1.4.10

### PR DESCRIPTION
**Pull request by Dexterp37**
Nothing surprising, but see the individual commits for more info.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 [Full changelog](https://github.com/mozilla/glean/compare/v33.0.0...main)
 
+* Android
+  * Update the Kotlin Gradle Plugin to version 1.4.10.
```
Comment (badboy) [inline CHANGELOG.md:6]: For my own understanding: this means we compile and test against that version of Kotlin and don't promise any lower Kotlin version still works.
But consumers _could_ compile their own code against other Kotlin versions and still rely on us?

File: glean-core/android/src/test/java/mozilla/telemetry/glean/private/EventMetricTypeTest.kt [hunk 2]
```diff
@@ -450,12 +450,12 @@ class EventMetricTypeTest {
         // Trigger worker task to upload the pings in the background
         triggerWorkManager(context)
 
-        var request = server.takeRequest(20L, TimeUnit.SECONDS)
-        var docType = request.path.split("/")[3]
+        val request = server.takeRequest(20L, TimeUnit.SECONDS)
+        val docType = request.path.split("/")[3]
```
Comment (badboy) [inline glean-core/android/src/test/java/mozilla/telemetry/glean/private/EventMetricTypeTest.kt:454]: what's the reason for this? no more mutability needed? or it was never needed, but it didn't produce warning/errors?

**Review by badboy — APPROVED** [review]
No blocker, just a couple of questions.

File: CHANGELOG.md (hunk 1 above)
Comment (Dexterp37) [reply CHANGELOG.md:6]: Yes, that's my understanding as well

File: glean-core/android/src/test/java/mozilla/telemetry/glean/private/EventMetricTypeTest.kt (hunk 2 above)
Comment (Dexterp37) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/private/EventMetricTypeTest.kt:454]: It was never needed, for some reason it did not produce a warning at compile time, but it produced warnings within Android Studio

File: CHANGELOG.md (hunk 1 above)
Comment (Dexterp37) [reply CHANGELOG.md:6]: I double checked with @pocmo and he confirmed this. This is also discussed in the [Kotlin Components Stability](https://kotlinlang.org/docs/reference/evolution/components-stability.html) page!

**Pull request merged**